### ISSUE TITLE
Auto scaling target tracking policies confused by cooldown

### ISSUE THREAD

**Issue by tilgovi** [terraform-providers/terraform-provider-aws#3689]
### Terraform Version
v0.11.3

### Affected Resource(s)
- aws_autoscaling_policy

### Terraform Configuration Files
```hcl
resource "aws_autoscaling_policy" "staging_memory" {
  name = "memory-reservation"

  autoscaling_group_name = "staging"
  policy_type            = "TargetTrackingScaling"

  target_tracking_configuration {
    customized_metric_specification {
      metric_dimension {
        name  = "ClusterName"
        value = "staging"
      }

      metric_name = "MemoryReservation"
      namespace   = "AWS/ECS"
      statistic   = "Maximum"
    }

    target_value = "75"
  }
}
```

### Expected Behavior
Terraform should be able to modify properties of the autoscaling policy.

### Actual Behavior
Terraform complains that target tracking policies cannot have a cooldown.
https://gist.github.com/tilgovi/75b45757ba6cd86c962c8adc60cce000

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Define a target tracking policy for an autoscaling group
2. `terraform apply`
3. Modify any property of the policy
4. `terraform apply`

### Important Factoids
It seems as though maybe the AWS API reports a default value that Terraform is then complaining about being non-nil.

**Comment by loivis** [comment]
From api specification, `cooldown` is not supported for `TargetTrackingScaling`. So the issue is that you can create but can't update, similar as #3715 ?

```
	// The amount of time, in seconds, after a scaling activity completes and before
	// the next scaling activity can start. If this parameter is not specified,
	// the default cooldown period for the group applies.
	//
	// This parameter is supported if the policy type is SimpleScaling.
	//
	// For more information, see Auto Scaling Cooldowns (http://docs.aws.amazon.com/autoscaling/latest/userguide/Cooldown.html)
	// in the Auto Scaling User Guide.
	Cooldown *int64 `type:"integer"`
```

**Comment by tilgovi** [comment]
That's correct. Oh, I realize my example is confusing. I added cooldown of zero as an experiment to try to trick the validation, but my actually code has no cooldown. Sorry about that.

**Comment by loivis** [comment]
np. I think it should also be fixed by the same pr.

**Comment by bflad** [comment]
#3739 has been merged into master and will release with v1.13.0 of the AWS provider, likely in about a week.

**Comment by bflad** [comment]
Ahead of schedule, this has been released in [version 1.13.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1130-march-28-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
